tests/utils/getClaimDetails.test.tsx

```diff
@@ -0,0 +1,92 @@
+import getClaimDetails, {
+  buildBenefitYear,
+  formatCurrency,
+  formatDate,
+  getProgramExtensionPair,
+  programExtensionPairs,
+  programExtensionPairType,
+  programTypeNames,
+} from '../../utils/getClaimDetails'
+
+// Test getProgramExtensionPair()
+describe('Converting ProgramType to user-facing strings', () => {
+  it('returns the correct string for each known ProgramType', () => {
+    for (const [id, pair] of Object.entries(programExtensionPairs)) {
+      const parts: programExtensionPairType = getProgramExtensionPair(programTypeNames[id])
+      expect(parts.programType).toBe(pair.programType)
+      expect(parts.extensionType).toBe(parts.extensionType)
+    }
+  })
+
+  it('throws an error if an unknown ProgramType is given', () => {
+    expect(() => {
+      getProgramExtensionPair('unknown')
+    }).toThrowError('Unknown Program Type')
+  })
+})
+
+// Test formatDate()
+describe('Formatting dates', () => {
+  it('displays the expected date string', () => {
+    const notFormatted = '2013-09-27T00:00:00'
+    const formatted = formatDate(notFormatted)
+    expect(formatted).toEqual('9/27/2013')
+  })
+})
+
+// Test buildBenefitYear()
+describe('Constructing the benefit year', () => {
+  it('results in a date range string', () => {
+    const range = buildBenefitYear('2013-09-27T12:34:33', '2021-10-28T05:33:34')
+    expect(range).toBe('9/27/2013 - 10/28/2021')
+  })
+})
+
+// Test formatCurrency()
+describe('Formatting dollar amounts', () => {
+  it('displays the correct string for amounts with cents', () => {
+    const withCents = '532.82'
+    expect(formatCurrency(withCents)).toEqual(`$${withCents}`)
+  })
+  it('displays the correct string for amounts without cents', () => {
+    const noCents = '500'
+    expect(formatCurrency(noCents)).toEqual(`$${noCents}.00`)
+  })
+  it('displays the correct string for zero dollars', () => {
+    const zero = '0'
+    expect(formatCurrency(zero)).toEqual('$0.00')
+  })
+  it('displays the correct string for negative amounts', () => {
+    const negative = '-40900'
+    expect(formatCurrency(negative)).toEqual('-$40,900.00')
+  })
+})
+
+// Test getClaimDetails()
+describe('Constructing the Claim Details object', () => {
+  it('returns all the expected values', () => {
+    // Mock ClaimDetailsResults
+    const rawDetails = {
+      programType: 'UI',
+      benefitYearStartDate: '2021-05-21T00:00:00',
+      benefitYearEndDate: '2022-05-20T00:00:00',
+      claimBalance: 100,
+      weeklyBenefitAmount: 25,
+      lastPaymentIssued: '2021-03-12T00:00:00',
+      lastPaymentAmount: 10,
+      monetaryStatus: 'active',
```

Comment:
I think I flagged this already, but these two inputs just quietly disappear!

Reply:
See https://github.com/cagov/ui-claim-tracker/pull/307#discussion_r669174902